psst-gui/src/ui/home.rs

```diff
@@ -15,22 +21,310 @@ use super::{
+}
+
+pub fn best_of_artists() -> impl Widget<AppState> {
+    Async::new(spinner_widget, loaded_results_widget, error_widget)
+        .lens(
+            Ctx::make(
+                AppState::common_ctx,
+                AppState::home_detail.then(HomeDetail::best_of_artists),
+            )
+            .then(Ctx::in_promise()),
+        )
+        .on_command_async(
+            LOAD_MADE_FOR_YOU,
+            |_| WebApi::global().best_of_artists(),
+            |_, data, q| data.home_detail.best_of_artists.defer(q),
+            |_, data, r| data.home_detail.best_of_artists.update(r),
+        )
+}
+
+pub fn your_shows() -> impl Widget<AppState> {
+    Async::new(spinner_widget, loaded_results_widget, error_widget)
+        .lens(
+            Ctx::make(
+                AppState::common_ctx,
+                AppState::home_detail.then(HomeDetail::your_shows),
+            )
+            .then(Ctx::in_promise()),
+        )
+        .on_command_async(
+            LOAD_MADE_FOR_YOU,
+            |_| WebApi::global().your_shows(),
+            |_, data, q| data.home_detail.your_shows.defer(q),
+            |_, data, r| data.home_detail.your_shows.update(r),
+        )
+}
+
+pub fn jump_back_in() -> impl Widget<AppState> {
+    Async::new(spinner_widget, loaded_results_widget, error_widget)
+        .lens(
+            Ctx::make(
+                AppState::common_ctx,
+                AppState::home_detail.then(HomeDetail::jump_back_in),
+            )
+            .then(Ctx::in_promise()),
+        )
+        .on_command_async(
+            LOAD_MADE_FOR_YOU,
+            |_| WebApi::global().jump_back_in(),
+            |_, data, q| data.home_detail.jump_back_in.defer(q),
+            |_, data, r| data.home_detail.jump_back_in.update(r),
+        )
+}
+
+pub fn shows_that_you_might_like() -> impl Widget<AppState> {
+    Async::new(spinner_widget, loaded_results_widget, error_widget)
+        .lens(
+            Ctx::make(
+                AppState::common_ctx,
+                AppState::home_detail.then(HomeDetail::shows_that_you_might_like),
+            )
+            .then(Ctx::in_promise()),
+        )
+        .on_command_async(
+            LOAD_MADE_FOR_YOU,
+            |_| WebApi::global().shows_that_you_might_like(),
+            |_, data, q| data.home_detail.shows_that_you_might_like.defer(q),
+            |_, data, r| data.home_detail.shows_that_you_might_like.update(r),
+        )
+}
+
+fn loaded_results_widget() -> impl Widget<WithCtx<MixedView>> {
+    Either::new(
+        |results: &WithCtx<MixedView>, _| {
+            results.data.artists.is_empty()
+                && results.data.albums.is_empty()
+                && results.data.playlists.is_empty()
+                && results.data.shows.is_empty()
+        },
+        Label::new("No results")
+            .with_text_size(theme::TEXT_SIZE_LARGE)
+            .with_text_color(theme::PLACEHOLDER_COLOR)
+            .padding(theme::grid(6.0))
+            .center(),
```

Comment:
When does this happen?

Reply:
AH, it doesn't! That is very true. (This was an oversight when copying the code from the search result widget!